ios/Errors.swift

```diff
@@ -1,55 +1,665 @@
-        let error = [
-            "code": errorCode,
-            "message": message
+    /// Wraps an error object in the standard error structure.
+    ///
+    /// - Parameters:
+    ///   - code: The React Native error code
+    ///   - nativeErrorCode: The native SDK error code
+    ///   - message: The error message
+    ///   - metadata: Platform-specific metadata
+    /// - Returns: A dictionary with the error wrapped under the "error" key
+    private class func stripeWrappedError(code: String, nativeErrorCode: String, message: String, metadata: [String: Any]) -> [String: Any] {
+        let error = stripeErrorObject(
+            code: code,
+            nativeErrorCode: nativeErrorCode,
+            message: message,
+            metadata: metadata
+        )
+        return [ErrorConstants.errorKey: error]
+    }
+    
+    /// Creates a base error object dictionary with standard fields.
+    ///
+    /// - Parameters:
+    ///   - name: The error name (defaults to "StripeError")
+    ///   - code: The React Native error code
+    ///   - nativeErrorCode: The native SDK error code
+    ///   - message: The error message
+    ///   - metadata: Platform-specific metadata
+    /// - Returns: A dictionary containing error properties
+    private class func stripeErrorObject(name: String = ErrorConstants.stripeErrorName, code: String, nativeErrorCode: String, message: String, metadata: [String: Any]) -> [String: Any] {
+        return [
+            ErrorConstants.nameKey: name,
+            ErrorConstants.messageKey: message.isEmpty ? ErrorConstants.unknownErrorMessage : message,
+            ErrorConstants.codeKey: code,
+            ErrorConstants.nativeErrorCodeKey: nativeErrorCode,
+            ErrorConstants.metadataKey: metadata
         ]
-        return ["error": error]
     }
-}
 
-func busyMessage(command: String, by busyCommand: String) -> String {
-    return "Could not execute \(command) because the SDK is busy with another command: \(busyCommand)."
+    /// Maps an NSError to the unified StripeError object structure.
+    ///
+    /// Extracts error information, API error details, refund information, underlying error,
+    /// and platform-specific metadata from the NSError.
+    ///
+    /// - Parameter nsError: The NSError to convert
+    /// - Returns: A dictionary containing the complete StripeError structure
+    class func mapToStripeErrorObject(nsError: NSError) -> [String: Any] {
+        let errorInfo = extractErrorInformation(from: nsError)
+        
+        var result = stripeErrorObject(
+            name: errorInfo.errorName,
+            code: errorInfo.code,
+            nativeErrorCode: errorInfo.nativeErrorCode,
+            message: nsError.localizedDescription,
+            metadata: [:]
+        )
+        
+        if errorInfo.isStripeError {
+            extractApiErrorToTopLevel(from: nsError, to: &result)
+            extractResponseObjectFromConfirmError(from: nsError, to: &result)
+        }
```

Comment:
Do all iOS StripeErrors contain an API Error? Asking because `extractApiErrorToTopLevel` always returns a APIError object. This behavior might be different from Android where APIError object isn't created and returned if the SDK doesn't return one.

Reply:
good catch 👍 that's correct, the keys will only be filled in if it was a proper API error so 

> If we can't find a stripeAPIErrorCode, then we won't add the ApiError key to the return structure

will work